material/#1_Using-Jupyter-Lab/#1_Using-Jupyter-Lab.ipynb

```diff
@@ -0,0 +1,666 @@
+    "This allows you to run parts of you code without having to rerun everything again (i.e. very useful if you have a long running simulation and want to style the plot of the results).\n",
+    "But it can also lead to problems, when we don't take this behavior into account when writing the code.\n",
+    "\n",
+    "#### Common Problems:\n",
+    "* **Broken Notebook** - \n",
+    "The most severe problem that can occure is that when you open your notebook again and want to run your code (`Run All Cells`), it fails at a specific cell telling you that something (i.e. function, variable) isn't defined.\n",
+    "The most common reason for that is that you deleted/renamed that 'something'/its cell in the last session or defined 'something' in a cell below the cell where it is first needed.  \n",
+    "Since the kernel still remembered/already knew it (deleting from the notebook $\\neq$ deleting from the kernel), everything worked fine, but when you started a new session this 'something', wasn't there when it was needed, since the cells are evaluated in order from top to bottom.\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "**Note:**\n",
+    "    \n",
+    "To solve ordering problems you can move cells in the notebook. \n",
+    "\n",
+    "</div>\n",
+    "\n",
+    "* **It doesn't change, even so it should** - \n",
+    "This is an error, which will lead to a **`Broken Notebook`** and often wasts quite some time to find. This mostly happens if you rename (`refacor`) a function/variable in one part of your code, but forgot to change it somewhere else, since the kernel remembers the old function/variable, changing the new one won't affect parts where the old name is still used. \n",
+    "\n",
+    "#### Solution\n",
+    "To make sure that the notebook runs properly even if you run it for the first time, you should **always** confirm that everything it is runnable with a fresh kernel **after refactoring** code or **before stopping to work on it**.\n",
+    "You can confirm this by running **`Restart Kernel and Run All Cells...`**.\n",
+    "\n",
+    "### Not rendered Plots\n",
+    "\n",
+    "By default plots aren't rendered in the output cells. To activate this for `matplotlib` (which is also the default backend for `pandas`) you need to add the magic command `%matplotlib inline` or `%matplotlib notebook`.\n",
+    "\n",
+    "For other libraries like `bokhe` or `ploty` there are special plugins needed to work in jupyter-lab, since they relay on extra javascript code, those plugin are called `jupyter extensions` ().\n",
```

Comment:
() ???

Reply:
wanted to add a link there but had no internet to get the link 😄